MoMMI/Modules/github.py

```diff
@@ -765,9 +766,82 @@ def gh_register_help() -> None:
             async with session.post(post) as resp:
                 await resp.text()
 
+#todo
+# support short label codes like qol, bugfix etc, so not search for labels literally
+# filter for emojicracy, just use emoji-modifiers to calc total value of issue, then choose between top ~10ish
+#   /repos/:owner/:repo/issues/:issue_number/reactions?content=+1 (or hooray, heart, confused, laugh, -1)
+#   needs Accept: application/json in header
+# make it possible to harddefine params eg. repo = bla/bla, so you don't have to give a label to search other repos
+#   would also be nice to have when the emojicracy-filter gets to be a thing
+# dont use \w
+@command("giveissue", r"giveissue(?:\s+(-\w+=\w+(?:\s+-\w+=\w+)*))?")
+async def giveissue_command(channel: MChannel, match: Match, message: Message) -> None:
+    try:
+        cfg: List[Dict[str, Any]] = channel.server_config("modules.github.repos")
+    except:
+        # Server has no config settings for GitHub.
+        await channel.send(":trash: No config found")
+        return
+
+    await channel.send(":hourglass_flowing_sand: Fetching random issue")
+
+    #default params
+    #repo = "vgstation-coders/vgstation13"
+    prefix = ""
+    labels = ""
+    emote = ""
+    ranking_limit = 20
+
+    
+    #getting params
+    text_params = [x.strip() for x in match.group(1).split(" -")] # strip whitespaces
+    
+    for param in text_params:
+        temp = param.split("=")
+        if temp[0] == "prefix":
+            prefix = temp[1].strip()
+            continue
+        if temp[0] == "labels":
+            labels = temp[1].strip()
+            continue
+        if temp[0] == "emote":
+            emote = re.search(REG_GIT_EMOTE, temp[1]).group(0)
+            continue
+        if temp[0] == "limit":
+            ranking_limit = int(temp[1])
+            continue
+        await channel.send(f":trash: Warning: Unknown parameter: {temp[0]}")
+
+    for repo_config in cfg:
+        repo = repo_config["repo"]
+
+        if not is_repo_valid_for_command(repo_config, channel, prefix):
+            continue
+
+        issues = []
+        i = 1
+        while 1:
+            url = github_url(f"/repos/{repo}/issues")
+            part_issues = await get_github_object(url, {"labels" : labels, "page" : i, "per_page" : "100"})
+            if part_issues:
+                issues += part_issues
+                i += 1
+            else:
+                break
+
+        if emote:
+            sort = sorted(issues, key=lambda i: get_github_object(f"{i.url}/reactions",{"content" : emote, "per_page" : "100"}).len)[ranking_limit:]
```

Comment:
You are doing a request for every issue, which will absolutely destroy the rate limit not to mention how horrendously long it would take.

Honestly you might just have to remove the emoji picker because I can't think of a "good" way to implement it.

Reply:
emoji picker overrated